core/src/main/scala/com/eharmony/spotz/backend/ParallelFunctions.scala

```diff
@@ -0,0 +1,82 @@
+package com.eharmony.spotz.backend
+
+import com.eharmony.spotz.objective.Objective
+import com.eharmony.spotz.optimizer.grid.Grid
+import com.eharmony.spotz.optimizer.hyperparam.RandomSampler
+
+import scala.reflect.ClassTag
+
+/**
+  * Use scala parallel collections to parallelize the optimizer computation.
+  */
+trait ParallelFunctions extends BackendFunctions {
+
+  /**
```

Comment:
You can probably move this scala doc and the next function into the parent class right?  Well most of it at least.